### PR TITLE
Fm 15 create test data for card deck

### PR DESCRIPTION
**Pull request by habibuga**
# Description

Created factory for generating test data for models of Deck, Category, Tag, Difficulty Level. Updated custom management command in users application to be able to generate test data for these object models.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Change was tested manually by generating test objects.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] The related memme has been added

alt memme

### CODE REVIEW

**Review by martynawitkowska — CHANGES_REQUESTED** [review]
cool, generating works, but tests need to be added for custom command options, and please add instructions in readme.md, because in the future not only backend will use this test data, but it is possible that also frontend will use it. Let's make it easier for them. A steb by step instuction please :D

File: flashcards/backend/decks/factories.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+from factory import Faker, Sequence, SubFactory
+from factory.django import DjangoModelFactory, ImageField
+from .models import Category, Deck, Flashcard, Tag, DifficultyLevel
```
Comment (ToJestKrzysio) [inline flashcards/backend/decks/factories.py:3]: PEP8 missing new line after lib imports
I'd consider absolute import in line 3

File: flashcards/backend/users/management/commands/create_test_data.py [hunk 2]
```diff
@@ -4,16 +4,17 @@
 from django.db import transaction
 
 from django.conf import settings
```
Comment (ToJestKrzysio) [inline flashcards/backend/users/management/commands/create_test_data.py:6]: [imports by PEP8](https://peps.python.org/pep-0008/#imports)
Please fix ordering of the imports, empty line in line 5? Shouldnt' it be after curremt line 6?

File: flashcards/backend/users/management/commands/delete_test_data.py [hunk 3]
```diff
@@ -2,11 +2,12 @@
 from django.db import transaction
 
 from django.conf import settings
+from decks.models import Category, Deck, DifficultyLevel, Tag
```
Comment (ToJestKrzysio) [inline flashcards/backend/users/management/commands/delete_test_data.py:5]: once again imports.

**Review by ToJestKrzysio — CHANGES_REQUESTED** [review]

File: flashcards/backend/users/management/commands/create_test_data.py [hunk 4]
```diff
@@ -29,7 +30,11 @@ def handle(self, *args, **kwargs):
         :param dict kwargs: kwargs
         :return: None
         """
-        objects_count = kwargs["objects_count"]
```
[the remark this answers is not in the document]
Comment (habibuga) [reply flashcards/backend/users/management/commands/create_test_data.py:32]: done, tests added for the command to create objects from all five models and README.MD is updated :smiley:

File: flashcards/backend/decks/factories.py (hunk 1 above)
Comment (habibuga) [reply flashcards/backend/decks/factories.py]: ok, now it's updated

File: flashcards/backend/users/management/commands/create_test_data.py (hunk 2 above)
Comment (habibuga) [reply flashcards/backend/users/management/commands/create_test_data.py]: ok, now it's updated

File: flashcards/backend/users/management/commands/delete_test_data.py (hunk 3 above)
Comment (habibuga) [reply flashcards/backend/users/management/commands/delete_test_data.py:5]: ok, now it's updated

**Review by gonzur75 — APPROVED** [review]
I've done manual test, and check the code and in my opinion it's all working fine.

**Review by martynawitkowska — CHANGES_REQUESTED** [review]
Black reformatted 3 files

**Review by ToJestKrzysio — APPROVED** [review]
No more problems

File: flashcards/readme.md [hunk 5]
```diff
@@ -229,10 +229,22 @@ For this reason suggested approach would be to install cypress locally and run i
    > 
 
 ## Test data generator
-1. To add some test data to database run:
-    > int_value = how many objects do you want to create
-    > 
-    >`docker compose exec <backend_service_name> python manage.py create_test_data int_value `
+1. To add some test data to the database run the command 
+    `docker compose exec <backend_service_name> python manage.py create_test_data` with the 
+    following optional attributes:
```
[the remark this answers is not in the document]
Comment (habibuga) [reply flashcards/readme.md]: files admin.py, models.py and migration 0001_initial.py updated by black
wording in readme.md changed to arguments :wink:

**Review by martynawitkowska — APPROVED** [review]
I would think about a higher range in difficulty level factory to avoid unique constraint error, in the end it is only test data. All else works fine. So I think we're good here for now.

**Pull request merged**